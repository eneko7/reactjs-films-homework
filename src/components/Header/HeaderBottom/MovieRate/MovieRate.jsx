import React from 'react';
import PropTypes from 'prop-types';
import style from './MovieRate.scss';

const MovieRate = (props) => {
  const {
    rate,
  } = props;
  const rateStyle = {
    width: `${(rate - 0.1) / 5 * 100}%`,
  };
  return (
    <div className={style.bottom_left_filmRate}>
      <div className={style.bottom_left_filmRate_stars_outer}>
        <div id="rate" style={rateStyle} className={style.bottom_left_filmRate_stars_outer_stars_inner} />
      </div>
      <div className={style.bottom_left_filmRate_number}>
        <span>{rate}</span>
      </div>
    </div>
  );
};

MovieRate.propTypes = {
  rate: PropTypes.string.isRequired,
};

export default MovieRate;
