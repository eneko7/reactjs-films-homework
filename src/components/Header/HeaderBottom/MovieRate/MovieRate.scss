.bottom_left_filmRate {
  width: 100%;
  margin-top: 25px;

  .bottom_left_filmRate_stars_outer {
    display: block;
    position: relative;
    float: left;

    .bottom_left_filmRate_stars_outer_stars_inner {
      position: absolute;
      top: 0;
      left: 0;
      white-space: nowrap;
      overflow: hidden;
      width: 0;
    }

    .bottom_left_filmRate_stars_outer_stars_inner::before {
      content: "★ ★ ★ ★ ★";
      color: #09ade3;
      font-size: 36px;
    }
  }

  .bottom_left_filmRate_stars_outer::before {
    content: "☆ ☆ ☆ ☆ ☆";
    color: #09ade3;
    font-size: 36px;
  }

  .bottom_left_filmRate_number {
    display: inline-block;
    font-size: 24px;
    line-height: 24px;
    color: #fff;
    font-family: Helvetica, Arial, Verdana, Tahoma, sans-serif;
    text-align: center;
    border-radius: 6px;
    border: 2px solid #09ade3;
    padding: 3px 5px;
    position: relative;
    top: 10px;
    left: 15px;
  }
}

@media (max-width: 767px) {
  .bottom_left_filmRate {
    margin-top: 10px;

    .bottom_left_filmRate_stars_outer {
      .bottom_left_filmRate_stars_outer_stars_inner::before {
        font-size: 25px;
      }
    }

    .bottom_left_filmRate_stars_outer::before {
      font-size: 25px;
    }

    .bottom_left_filmRate_number {
      font-size: 20px;
      line-height: 20px;
      top: 0;
    }
  }
}
