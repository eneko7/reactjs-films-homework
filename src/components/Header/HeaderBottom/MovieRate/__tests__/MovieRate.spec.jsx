import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MovieRate from '../MovieRate';

describe('MovieRate', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieRate rate="4.8" />);
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
