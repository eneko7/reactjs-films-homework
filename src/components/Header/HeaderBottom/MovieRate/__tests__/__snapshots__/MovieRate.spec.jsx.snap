// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieRate renders correctly 1`] = `
<div
  className="bottom_left_filmRate"
>
  <div
    className="bottom_left_filmRate_stars_outer"
  >
    <div
      className="bottom_left_filmRate_stars_outer_stars_inner"
      id="rate"
      style={
        Object {
          "width": "48.99999999999999%",
        }
      }
    />
  </div>
  <div
    className="bottom_left_filmRate_number"
  >
    <span>
      4.8
    </span>
  </div>
</div>
`;
